src/goe/offload/oracle/oracle_offload_transport_rdbms_api.py

```diff
@@ -344,26 +346,25 @@ def get_rdbms_query_cast(
             rdbms_column.data_type == ORACLE_TYPE_DATE
             and convert_expressions_on_rdbms_side
         ):
-            cast_expression = "TO_CHAR(%s,'YYYY-MM-DD HH24:MI:SS')" % column_expression
+            cast_expression = f"TO_CHAR({column_expression},'YYYY-MM-DD HH24:MI:SS')"
         elif (
             rdbms_column.data_type == ORACLE_TYPE_TIMESTAMP
             and convert_expressions_on_rdbms_side
         ):
-            cast_expression = "TO_CHAR(%s,'YYYY-MM-DD HH24:MI:SS.FF%s')" % (
-                column_expression,
-                ff_scale,
+            cast_expression = (
+                f"TO_CHAR({column_expression},'YYYY-MM-DD HH24:MI:SS.FF{ff_scale}')"
             )
         elif (
             staging_column.is_string_based()
             and rdbms_column.data_type in (ORACLE_TYPE_NUMBER, ORACLE_TYPE_FLOAT)
             and convert_expressions_on_rdbms_side
         ):
-            cast_expression = "TO_CHAR(%s,'TM')" % column_expression
+            cast_expression = f"TO_CHAR({column_expression},'TM')"
         elif (
             rdbms_column.data_type in (ORACLE_TYPE_INTERVAL_DS, ORACLE_TYPE_INTERVAL_YM)
             and for_spark
-        ):
-            cast_expression = "TO_CHAR(%s)" % column_expression
+        ) or (rdbms_column.data_type == ORACLE_TYPE_INTERVAL_YM and for_qi):
```

Comment:
If intervals are converted to string, why do we differentiate between supporting DS and YM for Spark and only YM for Query Import?

Reply:
It's based on what cx-Oracle supports. We get the exception below when trying to query interval YM:
```
    ora_cursor.execute(source_query)
cx_Oracle.NotSupportedError: Oracle type 2016 not supported.
```
But interval DS is supported and is currently offloaded. So I was just plugging the gap in cx-Oracle support.

Reply:
We can simplify this to applying the TO_CHAR cast expression to DS and YM for both Spark and Qi.

Reply:
Agreed, I've made the change.